ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryRegistryImpl.java

```diff
@@ -333,8 +333,13 @@ private void registerPersistentQuery(
       unregisterQuery(oldQuery);
     }
 
-    // Initialize the query before it's exposed to other threads via the map/sets.
-    persistentQuery.initialize();
+    // If the old query was sandboxed, then the stop() won't stop the streams and will cause
```

Comment:
For my understanding, can you explain why `stop()` won't stop streams? Or maybe more broadly, why we were running these KStreams applications in a sandboxed environment? Just linking me to something would be sufficient, I wasn't able to piece it together myself easily.